File: Archive/entries_helper.py
import Entry

def add_entry(logs): # logs is a list of log entries
    # still need to fill in the information retrieval
    date = None
    dist = None
    loc = None
    purp = None
    logs.append(Entry(date, dist, loc, purp))

def rm_entry(logs, entry): # entry is an index
    logs.pop(entry)

def edit_entry(logs, entry):
    date = None
    dist = None
    loc = None
    purp = None
    if date != None:
        logs[entry].edit_date(date)
    if dist != None:
        logs[entry].edit_dist(dist)
    if loc != None:
        logs[entry].edit_loc(loc)
    if purp != None:
        logs[entry].edit_purp(purp)

File: entries_helper.js
import Entry from "Entry.js";

const newEntryDataDiv = document.getElementsByClassName('.newEntryData');
const newEntryButDiv = document.getElementById('newEntryBut');
const updateEntryDataDiv = document.getElementsByClassName('.updateEntryData');
const updateEntryButDiv = document.getElementsByClassName('updateEntryBut');

const addEntry = (log) => {
    let date = newEntryDataDiv[0].value;
    let dist = newEntryDataDiv[1].value;
    let loc = newEntryDataDiv[2].value;
    let purp = newEntryDataDiv[3].value;
    let new_entry = new Entry(date, dist, loc, purp);
    log.push(new_entry);
}

const rmEntry = (log, entry) => {
    log.pop(entry);
}

const editEntry = (log, entry, editType) => {
    
}

const loadLog = () => {

}

const saveLog = () => {

}

const tempAutosave = () => {

}

const miToKm = () => {

}

const kmToMi = () => {
    
}